src/commonMain/kotlin/exchange.dydx.abacus/state/manager/notification/providers/BlockRewardNotificationProvider.kt

```diff
@@ -0,0 +1,87 @@
+package exchange.dydx.abacus.state.manager.notification.providers
+
+import exchange.dydx.abacus.output.BlockReward
+import exchange.dydx.abacus.output.Notification
+import exchange.dydx.abacus.output.NotificationPriority
+import exchange.dydx.abacus.output.NotificationType
+import exchange.dydx.abacus.state.manager.V4Environment
+import exchange.dydx.abacus.state.manager.notification.NotificationsProviderProtocol
+import exchange.dydx.abacus.state.model.TradingStateMachine
+import exchange.dydx.abacus.utils.IMap
+import exchange.dydx.abacus.utils.JsonEncoder
+import exchange.dydx.abacus.utils.UIImplementations
+import exchange.dydx.abacus.utils.iMapOf
+import exchange.dydx.abacus.utils.typedSafeSet
+import kollections.toIMap
+
+class BlockRewardNotificationProvider(
+    private val uiImplementations: UIImplementations,
+    private val environment: V4Environment,
+    private val jsonEncoder: JsonEncoder,
+) : NotificationsProviderProtocol {
+    override fun buildNotifications(
+        stateMachine: TradingStateMachine,
+        subaccountNumber: Int
+    ): IMap<String, Notification> {
+        /*
+           We have to go through fills instead of orders, because
+           1. Order doesn't have an updatedAt timestamp
+           2. Order doesn't have an average filled price
+         */
+        val account =
+            stateMachine.state?.account ?: return kollections.iMapOf()
+
+        val notifications = exchange.dydx.abacus.utils.mutableMapOf<String, Notification>()
+        val accountBlockRewards = account.tradingRewards?.blockRewards
+        val token = environment.tokens["chain"]?.name
+        if (accountBlockRewards != null && token != null) {
+            for (blockReward in accountBlockRewards) {
+                createBlockRewardNotification(stateMachine, blockReward, token)?.let {
+                    notifications.typedSafeSet(
+                        it.id,
+                        it,
+                    )
+                }
+            }
+        }
+        return notifications
+    }
+
+    private fun createBlockRewardNotification(
+        stateMachine: TradingStateMachine,
```

Comment:
we're not using this param?

Reply:
The code is copy-paste from old NotificationsProvider.  But we can delete this.